core/src/replay_stage.rs

```diff
@@ -832,6 +863,83 @@ impl ReplayStage {
         }
     }
 
+    fn retransmit_not_propagated(
+        poh_recorder: &Arc<Mutex<PohRecorder>>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+        retransmit_slots_sender: &RetransmitSlotsSender,
+        progress: &mut ProgressMap,
+        last_retransmit_retry_info: &mut LastRetransmitInfo,
+    ) -> Option<Slot> {
+        let mut retransmitted_slot = None;
+        let start_slot = poh_recorder.lock().unwrap().start_slot();
+
+        if let Some(latest_leader_slot) = progress.get_latest_leader_slot(start_slot) {
```

Comment:
While reading this, this reminded me of an issue I'd thought of earlier. Created the issue here: https://github.com/solana-labs/solana/issues/21659

Reply:
makes sense, I'll address that in #21659